src/java.base/share/classes/java/lang/reflect/ProxyGenerator.java

```diff
@@ -714,100 +723,72 @@ private ProxyMethod(Method method, String sig, Class<?>[] parameterTypes,
          * @param method          The method for which to create a proxy
          * @param methodFieldName the fieldName to generate
          */
-        private ProxyMethod(Method method, String methodFieldName) {
+        private ProxyMethod(Method method, Utf8Entry methodFieldName) {
             this(method, method.toShortSignature(),
-                    method.getSharedParameterTypes(), method.getReturnType(),
-                    method.getSharedExceptionTypes(), method.getDeclaringClass(), methodFieldName);
+                 method.getSharedParameterTypes(), method.getReturnType(),
+                 method.getSharedExceptionTypes(), method.getDeclaringClass(), methodFieldName);
         }
 
         /**
          * Generate this method, including the code and exception table entry.
          */
-        private void generateMethod(ClassWriter cw, String className) {
-            MethodType mt = MethodType.methodType(returnType, parameterTypes);
-            String desc = mt.toMethodDescriptorString();
-            int accessFlags = ACC_PUBLIC | ACC_FINAL;
-            if (method.isVarArgs()) accessFlags |= ACC_VARARGS;
-
-            MethodVisitor mv = cw.visitMethod(accessFlags,
-                    method.getName(), desc, null,
-                    typeNames(Arrays.asList(exceptionTypes)));
-
-            int[] parameterSlot = new int[parameterTypes.length];
-            int nextSlot = 1;
-            for (int i = 0; i < parameterSlot.length; i++) {
-                parameterSlot[i] = nextSlot;
-                nextSlot += getWordsPerType(parameterTypes[i]);
+        private void generateMethod(ClassBuilder clb, ClassEntry className) {
+            var cp = clb.constantPool();
+            var desc = new StringJoiner("", "(", ")" + returnType.descriptorString());
+            for (var pt : parameterTypes) {
+                desc.add(pt.descriptorString());
```

Comment:
Maybe worth refactor these as `ProxyMethod::toMethodTypeDescriptorString` with the comment to explain why `MethodType::descriptorString` is not used.

Reply:
I've decided to revert this custom optimization.